pandas/tests/io/formats/test_format.py

```diff
@@ -239,6 +239,18 @@ def test_repr_truncation(self):
         with option_context("display.max_colwidth", max_len + 2):
             assert "..." not in repr(df)
 
+    def test_repr_deprecation_negative_int(self):
+        # FIXME: remove in version 1.2 after deprecation cycle
+        # Non-regression test for:
+        # https://github.com/pandas-dev/pandas/issues/31532
+        with pytest.warns(FutureWarning) as records:
```

Comment:
Thanks!

Could you please use
```
with tm.assert_produces_warning(
```
instead? See [testing warnings](https://pandas.pydata.org/pandas-docs/stable/development/contributing.html#testing-warnings)